### DISCUSSION TITLE
(Q&A and Idea) How do i set html default value in Vue dom?

### DISCUSSION THREAD

**Discussion by Nemure231** [Milkdown/milkdown#985] [Q&A] [answered]
Hi, i was planed to add milkdown it in vue/nuxt base project with html value. But i can't set the html value by just following the documentation example. How can i set the html value?

And i was thinking maybe this html value works in vue dom by wraping the editor in onMounted, but its still doesn't work. 
![Screenshot 2023-04-22 225043](https://user-images.githubusercontent.com/49015080/233794019-cdac9dc8-eed8-4ca3-b0d9-41cb24c623a7.jpg)

Anyway, i also have idea about this. It's will be cool if we can write html value inside markdown with vue slots like this:

```HTML
<Milkdown>
    <template #htmlValue>
           <div>
             <h1>Hello milkdown!</h1>
           </div>
     </template>
</Milkdown>
```

**Comment by Saul-Mirone** [comment, marked as answer]
The dom hasn't been rendered for vue in your case, move the markdown variable into `onMounted`:

```html
<script setup lang="ts">
import { Milkdown, useEditor } from '@milkdown/vue';
import { defaultValueCtx, Editor, rootCtx } from '@milkdown/core';
import { nord } from '@milkdown/theme-nord'
import { commonmark } from '@milkdown/preset-commonmark'

useEditor((root) => {
  return Editor.make()
    .config(nord)
    .config((ctx) => {
      ctx.set(rootCtx, root)
      ctx.set(defaultValueCtx, {
        type: 'html',
        dom: document.querySelector('#pre'),
      })
    })
    .use(commonmark)
})
</script>

<template>
  <div id="pre">
    <h1>Milkdown Vue Commonmark</h1>
  </div>
  <Milkdown />
</template>
```

**Reply by Nemure231** [reply]
wow thank you, its really works!
